src/lib/containers/drug_upload_tool/UserFileGrid.tsx

```diff
@@ -0,0 +1,269 @@
+      this.setState({
+        userprofile,
+        dataFolderId: result.dataFolderId,
+        fileList: result.fileList || [],
+        isLoading: false
+      })
+      return result;
+    }
+  }
+  onError = (args: any) => {
+    console.log(args)
+  }
+
+
+  // creates data folder for user
+  createDataFolder = async (userprofile: UserProfile, token: string): Promise<string | undefined> => {
+
+    const newFolder: Entity = {
+      parentId: this.props.parentContainerId,
+      name: userprofile.ownerId,
+      concreteType: 'org.sagebionetworks.repo.model.Folder',
+    }
+    try {
+      const folder: Entity = await SynapseClient.createEntity(newFolder, token);
+      return folder.id
+    }
+    catch (error) {
+      this.onError(error)
+      throw (error)
+    }
+  }
+
+
+
+  getFileListingInFolder = async (token: string, targetFolderId: string): Promise<Entity[]> => {
+    const request = {
+      includeTypes: ['file'],
+      parentId: targetFolderId,
+      sortBy: 'NAME',
+      sortDirection: 'ASC'
+    }
+
+    try {
+      const data = await SynapseClient.getEntityChildren(request, token);
+      return data.page;
+    }
+    catch (error) {
+      this.onError(error)
+      return [];
+    }
+  }
+
+  deleteFile = async (token: string, entityId: string): Promise<any> => {
+
+    this.setState({
+      isLoading: true, modalContext: undefined
+    })
+
+    try {
+      await SynapseClient.deleteEntity(token, entityId);
+      const fileList = await this.getFileListingInFolder(token, this.state.dataFolderId!)
+      this.setState({
+        fileList
+      })
+    }
+    catch (error) {
+      this.onError(error)
+      return [];
+    }
+    finally {
+      this.setState({
+        isLoading: false
+      })
+    }
+  }
+
+  /* ------------------------------------------   rendering fns  ------------------------------------------------*/
+ 
+  renderLoading = (token: string | undefined, isLoading: boolean): JSX.Element => {
+    if (
+      token &&
+      isLoading) {
+      return (<div className='text-center'>
+        <span>Loading&hellip;</span>
+        <span style={{ marginLeft: '2px' }} className={'spinner'} />
+      </div>)
+    } else {
+      return <></>
+    }
+  }
+
+  renderFileTable = (fileList: Entity[], pathpart: string, dataFolderId?: string): JSX.Element => {
+    if (!dataFolderId) {
+      this.onError('Data Folder ID is undefined')
+      return <></>
+    }
+    if (_.isEmpty(fileList)) {
+      return <h5 className='text-center no-submissions padding-full'>You don't have submissions in progress [COPY!]</h5>
+    }
+    return (<table className='table file-table' >
+      <thead>
+        <tr>
+          <th>Submission Name</th>
+          <th>Edited On</th>
+          <th>Submitted On</th>
+          <th>Action</th>
+
+        </tr>
+      </thead>
+      <tbody>
+        {fileList.map((entity, i: number) => {
+          return (<tr className='test' key={i}>
```

Comment:
For key prop you could use `entity.id` to have a stable key

Reply:
out of curiosity, why does it matter?

Reply:
If I understand this tool/interaction correctly, this _shouldn't_ really matter with what will happen, but we've had some pretty big bugs in the past with things re-rendering a lot because of setting the key to an index or random number (it was in a tutorial online, literally the worst thing you could set the key prop to be..) so we just always use a stable key so react doesn't re-render and drop the page/content because it thinks something new is on the screen.